src/components/fabSpeedDial/fabController.js

```diff
@@ -26,7 +26,20 @@
       $scope.$evalAsync("vm.isOpen = !vm.isOpen");
     };
 
-    setupDefaults();
+    /*
+     * Angular Lifecycle hook for newer Angular versions.
+     * Bindings are no longer assigned in the controller, but in the $onInit hook.
+     */
+    vm.$onInit = function() {
+      setupDefaults();
+    };
+
+    // For Angular 1.4 and older, where there are no lifecycle hooks but bindings are pre-assigned,
+    // manually call the $onInit hook.
+    if (angular.version.major === 1 && angular.version.minor <= 4) {
+      this.$onInit();
+    }
```

Comment:
It feels a little odd to call `$onInit()` before the end of the controller initialization. It will also make the difference between 1.4 vs 1.6 more significant. AFAICT, it doesn't matter with the current implementation, but it makes the code more brittle (e.g. a change inside `setupListeners/Watchers()` or `fireInitialAnimations()` might have unexpected results.

Was there a specific reason for not moving all initialization inside `$onInit()` or not at least calling `$onInit()` at the end of the controller constructor?